core/base/inc/TVirtualMutex.h

```diff
@@ -58,9 +58,12 @@ class TVirtualMutex {
 //    TLockGuard guard(mutex);                                          //
 //    ... // do something                                               //
 // }                                                                    //
-// when guard goes out of scope the mutex is unlocked in the TLockGuard //
+// where mutex is a pointer to a TMutex object.                         //
+// When guard goes out of scope the mutex is unlocked in the TLockGuard //
 // destructor. The exception mechanism takes care of calling the dtors  //
 // of local objects so it is exception safe.                            //
+// In contrast to std::lock_guard, TLockGuard constructor expects a     //
+// pointer, not the mutex object itself, to support polymorphism.       //
```

Comment:
```suggestion
// pointer, not the mutex object itself.                                //
```
(Polymorphism through reference works just as well. We really don't have a good reason except for uniformity of the legacy interfaces.)